### PR TITLE
feat/new typography styles

### PR DESCRIPTION
**Pull request by gui-santos**
# Purpose of PR

Screenshot 2021-02-04 at 14 11 24

It updates font-sizes, line-height, and weights in all typography
And It creates the new huge size for DisplayText
I also made the overview look like the Figma file

I got inspired by [SumUp’s CircuitUI docs](https://circuit.sumup.com/?path=/story/typography-heading--sizes) and created a new section in storybook for our typography (see image above)
but in case you prefer to leave it in Components, I can change this back

P.S: Icon size change in Note component
Screenshot 2021-02-05 at 15 40 51

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by denkristoffer — APPROVED** [review]
I like the idea of splitting up the stories a bit more 👍

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const fontWeight = {
   'font-weight-normal': 400,
-  'font-weight-medium': 600,
+  'font-weight-medium': 500,
```
Comment (denkristoffer) [inline packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: This is a breaking change 🙀

**Review by denkristoffer — CHANGES_REQUESTED** [review]
I like the idea of splitting up the stories a bit more 👍

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js (hunk 1 above)
Comment (gui-santos) [reply packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: hmmmm and that's the most important part for me hahaha

@mshaaban0 @fabe and @burakukula why didn't you say in slack that we didn't have breaking changes in Typography?

What do we do now, friends?

Comment (denkristoffer) [reply packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: If we change our typography it's probably not breaking, but the token change is a breaking change IMO.

But we should align with https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#common_weight_name_mapping

File: packages/forma-36-tokens/src/tokens/typography/line-height.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
 const lineHeight = {
   'line-height-default': '1.5',
   'line-height-condensed': '1.25',
+  'line-height-4xl': '3.5rem',
```
Comment (domarku) [inline packages/forma-36-tokens/src/tokens/typography/line-height.js:4]: I see no reason to have more than two different line-heights. We're introducing unnecessary complexity.

Heading and Display should use the condensed one (1.25).
All the rest should use the default one (1.5).

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js (hunk 1 above)
Comment (burakukula) [reply packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: @denkristoffer @gui-santos change like it is done in here it's not a breaking change - we adjust the value of the token, not changing the namespace for the token itself. 
If we decide to change tokens namespaces, then yeah, it's v4, but I am not sure if we want to do it tbh. 
I am ok with this adjustment like it is for now.

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
   'font-weight-demi-bold': 700,
```
Comment (domarku) [inline packages/forma-36-tokens/src/tokens/typography/font-weight.js:4]: Change from 700 to 600

File: packages/forma-36-react-components/src/components/Typography/SectionHeading/SectionHeading.css [hunk 4]
```diff
@@ -5,10 +5,10 @@
 .SectionHeading {
   display: block;
   font-family: var(--font-stack-primary);
-  font-weight: var(--font-weight-demi-bold);
+  font-weight: var(--font-weight-medium);
```
Comment (domarku) [inline packages/forma-36-react-components/src/components/Typography/SectionHeading/SectionHeading.css:8]: This should also be demi-bold

File: packages/forma-36-react-components/src/components/Typography/Subheading/Subheading.css [hunk 5]
```diff
@@ -5,9 +5,9 @@
 .Subheading {
   display: block;
   font-family: var(--font-stack-primary);
-  font-weight: var(--font-weight-demi-bold);
+  font-weight: var(--font-weight-medium);
```
Comment (domarku) [inline packages/forma-36-react-components/src/components/Typography/Subheading/Subheading.css:8]: Keep it demi-bold

File: packages/forma-36-react-components/src/components/Typography/DisplayText/DisplayText.css [hunk 6]
```diff
@@ -5,17 +5,23 @@
 .DisplayText {
   display: block;
   font-family: var(--font-stack-primary);
-  font-weight: var(--font-weight-demi-bold);
+  font-weight: var(--font-weight-medium);
```
Comment (domarku) [inline packages/forma-36-react-components/src/components/Typography/DisplayText/DisplayText.css:8]: Keep it demi-bold

File: packages/forma-36-react-components/src/components/Typography/Heading/Heading.css [hunk 7]
```diff
@@ -5,10 +5,10 @@
 .Heading {
   display: block;
   font-family: var(--font-stack-primary);
-  font-weight: var(--font-weight-demi-bold);
+  font-weight: var(--font-weight-medium);
```
Comment (domarku) [inline packages/forma-36-react-components/src/components/Typography/Heading/Heading.css:8]: Perhaps leave the headings demi-bold

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js (hunk 1 above)
Comment (domarku) [reply packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: Changing the value here doesn't seem like a breaking change to me

**Review by domarku — COMMENTED** [review]
Make sure to adjust the hacky line-height in note and notification. @burakukula might know more locations
https://github.com/contentful/forma-36/blob/master/packages/forma-36-react-components/src/components/Note/Note.css#L24

File: packages/forma-36-tokens/src/tokens/typography/font-weight.js (hunk 1 above)
Comment (denkristoffer) [reply packages/forma-36-tokens/src/tokens/typography/font-weight.js:3]: After giving this a second thought I agree that this is no more breaking than all the other token changes we've done recently. I feel strongly about aligning with the normal weight names though 🙂

**Review by denkristoffer — APPROVED** [review]

**Comment by denkristoffer** [comment]
We should publicise the token change like with did with the others recently 👍

**Review by burakukula — APPROVED** [review]
🏆

**Review by fabe — APPROVED** [review]

**Pull request merged**